qiskit/providers/ibmq/managed/managedjobset.py

```diff
@@ -87,14 +104,88 @@ def run(
-            self._managed_jobs.append(
-                ManagedJob(experiments, start_index=exp_index,
-                           qobj=qobj, job_name=job_name, backend=backend,
-                           executor=executor, job_share_level=job_share_level,
-                           job_tags=job_tags)
-            )
+            mjob = ManagedJob(experiments_count=len(experiments), start_index=exp_index)
+            mjob.submit(qobj=qobj, job_name=job_name, backend=backend,
+                        executor=executor, job_share_level=job_share_level,
+                        job_tags=self._tags+[self._id_long])
+            self._managed_jobs.append(mjob)
             exp_index += len(experiments)
 
+    def retrieve_jobs(self, provider: AccountProvider, refresh: bool = False) -> None:
+        """Retrieve previously submitted jobs for this set.
+
+        Args:
+            provider: Provider used for this job set.
+            refresh: If True, re-query the API for the job set.
+                Otherwise return the cached value. Default: False.
+
+        Raises:
+            IBMQJobManagerUnknownJobSet: If the job set cannot be found.
+            IBMQJobManagerInvalidStateError: If jobs for this job set are
+                found but have unexpected attributes.
+        """
+        if not refresh and self._managed_jobs:
+            return
+
+        # IBMQBackend jobs() method does not have a way to pass in unlimited
+        # number of jobs to retrieve. 1000 should be a sufficiently large
+        # enough number.
+        jobs = []  # type: List[IBMQJob]
+        page_limit = 1000
+        while True:
+            job_page = provider.backends.jobs(    # type: ignore[attr-defined]
+                skip=len(jobs), limit=page_limit, job_tags=[self._id_long])
+            jobs += job_page
+            if len(job_page) < page_limit:
+                break
+
+        if not jobs:
+            raise IBMQJobManagerUnknownJobSet(
+                "{} is not a known job set within the provider {}.".format(
+                    self.job_set_id(), provider))
+
+        # Extract common information from the first job.
+        first_job = jobs[0]
+        pattern = re.compile(r'(.*)_([0-9])+_$')
+        matched = pattern.match(first_job.name())
+        if not matched:
+            raise IBMQJobManagerInvalidStateError(
+                "Job {} is tagged for the job set {} but does not have a proper job name.".format(
+                    first_job.job_id(), self.job_set_id()))
+        self._name = matched.group(1)
+        self._backend = first_job.backend()
+        self._tags = first_job.tags()
+        self._tags.remove(self._id_long)
+
+        jobs_dict = {}
+        for job in jobs:
+            # Verify the job is proper.
+            matched = pattern.match(job.name()) if job.name() else None
+            if not matched or matched.group(1) != self._name or \
+                    job.backend().name != self._backend.name:
```

Comment:
You must have forgot to call the `name()` method: `job.backend().name() != self._backend.name()` 😅

Reply:
Good catch!!!